bundle/src/main/java/com/adobe/acs/commons/properties/impl/PropertyAggregatorServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+                if (!"class".equalsIgnoreCase(propertyDescriptor.getName())) {
+                    String newKey = key + "." + propertyDescriptor.getName();
+                    Object propertyValue = propertyDescriptor.getReadMethod().invoke(beanValue);
+                    if (propertyValue != null) {
+                        targetPropertyMap.put(newKey, propertyValue);
+                    }
+                }
+            }
+
+        } catch (ClassNotFoundException e) {
+            log.error("Error mapping to class " + beanValue.getClass().getName(), e);
+        } catch (IntrospectionException e) {
+            log.error("IntrospectionException", e);
+        } catch (IllegalAccessException e) {
+            log.error("IllegalAccessException", e);
+        } catch (InvocationTargetException e) {
+            log.error("InvocationTargetException", e);
+        }
+    }
+
+    /**
+     * Add the properties of a page to the given map.  Excluded properties are found in the configuration for this service.
+     * @param targetPropertyMap the map that should be updated with the properties and their values
+     * @param page the page containing properties
+     */
+    private void addPagePropertiesToMap(final Map<String,Object> targetPropertyMap, final Page page, final String prefix) {
+        ValueMap pageProperties = page.getProperties();
+        Set<Map.Entry<String, Object>> entries = pageProperties.entrySet();
+        for (Map.Entry<String, Object> entry : entries) {
+            if (notInMap(targetPropertyMap, entry.getKey())
+                    && isNotExcluded(entry.getKey())
+                    && isAllowedType(pageProperties.get(entry.getKey()))) {
+                targetPropertyMap.put(prefix(prefix, entry.getKey()), pageProperties.get(entry.getKey()));
+            }
+        }
+    }
+
+    /**
+     * Recursively traverse the properties of a page and it's parents until that parent is not a page or is null, which
+     * means it's the root, and add them to the given map.  The traversing will stop when the parent is a node that is
+     * not a page, such as a folder.  This is necessary to support things such as editable templates under `/conf`.
+     *
+     * @param targetPropertyMap the map that should be updated with the properties and their values
+     * @param page the page containing properties
+     */
+    private void addInheritedPagePropertiesToMap(final Map<String,Object> targetPropertyMap, final Page page) {
+
+        addPagePropertiesToMap(targetPropertyMap, page, INHERITED_PAGE_PROP_PREFIX);
+
+        Resource pageResource = page.adaptTo(Resource.class);
+        if (pageResource != null) {
+            Resource parentResource = pageResource.getParent();
+
+            if (parentResource != null && NameConstants.NT_PAGE.equalsIgnoreCase(parentResource.getResourceType())) {
+                addInheritedPagePropertiesToMap(targetPropertyMap, parentResource.adaptTo(Page.class));
```

Comment:
You allow here that the values of the grand-parent page overwrite the values of the parent-page. Is that intended? I would expect that the parent page can overwrite properties from the grand-parent page in the targetPropertyMap.

Reply:
Check out lines 75-77, that's the priority for the addition. page properties get added first and then the method `notInMap` on line 189 makes sure it doesn't overwrite values.